File: Samples/FileSearch/js/js/scenario1.js
//// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF
//// ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO
//// THE IMPLIED WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A
//// PARTICULAR PURPOSE.
////
//// Copyright (c) Microsoft Corporation. All rights reserved

(function () {
    "use strict";

    var page = WinJS.UI.Pages.define("/html/scenario1.html", {
        ready: function (element, options) {
            document.getElementById("search").addEventListener("click", search, false);
        }
    });

    function search() {
        // Initiate the div elements
        var outputDiv = document.getElementById("output");
        var queryBox = document.getElementById("queryBox");

        // Retrieve the query entered in the textbox
        var searchFilter = queryBox.value;

        // Create a new file query from the music library and apply the AQS filter
        Windows.Storage.KnownFolders.getFolderForUserAsync(null /* current user */, Windows.Storage.KnownFolderId.musicLibrary).then(function (musicLibrary) {
            var options = new Windows.Storage.Search.QueryOptions(Windows.Storage.Search.CommonFileQuery.orderBySearchRank, ["*"]);
            options.userSearchFilter = searchFilter;
            var fileQuery = musicLibrary.createFileQueryWithOptions(options);
            return fileQuery.getFilesAsync();
        })
        .done(function (files) {
            if (files.size === 0) {
                outputDiv.innerText = "No files found for '" + searchFilter + "'";
            } else {
                // Create an output string to hold results count and filenames
                var filesLabel = (files.size === 1) ? "file" : "files";
                var output = files.size + " " + filesLabel + " found\n\n";

                // Iterate through the results and print each filename to the output field
                files.forEach(function (file) {
                    output += file.name + "\n";
                });
                outputDiv.innerText = output;
            }
        });
    }
})();

File: Samples/FileSearch/cppwinrt/Scenario1.cpp
//*********************************************************
//
// Copyright (c) Microsoft. All rights reserved.
// This code is licensed under the MIT License (MIT).
// THIS CODE IS PROVIDED *AS IS* WITHOUT WARRANTY OF
// ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING ANY
// IMPLIED WARRANTIES OF FITNESS FOR A PARTICULAR
// PURPOSE, MERCHANTABILITY, OR NON-INFRINGEMENT.
//
//*********************************************************

#include "pch.h"
#include "Scenario1.h"
#include "Scenario1.g.cpp"
#include <sstream>

using namespace winrt;
using namespace Windows::Foundation;
using namespace Windows::Foundation::Collections;
using namespace Windows::Storage;
using namespace Windows::Storage::Search;
using namespace Windows::UI::Xaml;

namespace winrt::SDKTemplate::implementation
{
    Scenario1::Scenario1()
    {
        InitializeComponent();
    }

    fire_and_forget Scenario1::SearchButton_Click(IInspectable const&, RoutedEventArgs const&)
    {
        auto lifetime = get_strong();

        StorageFolder musicFolder = co_await KnownFolders::GetFolderForUserAsync(nullptr /* current user */, KnownFolderId::MusicLibrary);

        QueryOptions queryOptions(CommonFileQuery::OrderBySearchRank, { L"*" });
        //use the user's input to make a query
        queryOptions.UserSearchFilter(InputTextBox().Text());
        StorageFileQueryResult queryResult = musicFolder.CreateFileQueryWithOptions(queryOptions);

        std::wstringstream output;

        //find all files that match the query
        IVectorView<StorageFile> files = co_await queryResult.GetFilesAsync();
        //output how many files that match the query were found
        if (files.Size() == 0)
        {
            output << L"No files found for '" << std::wstring_view{ queryOptions.UserSearchFilter() } << L"'";
        }
        else if (files.Size() == 1)
        {
            output << files.Size() << L" file found:" << std::endl << std::endl;
        }
        else
        {
            output << files.Size() << L" files found:" << std::endl << std::endl;
        }

        //output the name of each file that matches the query
        for (StorageFile const& file : files)
        {
            output << std::wstring_view{ file.Name() } << std::endl;
        }

        OutputTextBlock().Text(output.str());
    }
}
